pageserver/src/tenant/ephemeral_file/page_caching.rs

```diff
@@ -49,6 +58,46 @@ impl RW {
         self.rw.bytes_written()
     }
 
+    /// Load all blocks that can be read via [`Self::read_blk`] into a contiguous memory buffer.
+    ///
+    /// This includes the blocks that aren't yet flushed to disk by the internal buffered writer.
+    /// The last block is zero-padded to [`PAGE_SZ`], so, the returned buffer is always a multiple of [`PAGE_SZ`].
+    pub(super) async fn load_into_contiguous_memory(
+        &self,
+        ctx: &RequestContext,
+    ) -> Result<Vec<u8>, io::Error> {
+        // round up to the next PAGE_SZ multiple, required by blob_io
+        let size = {
+            let s = usize::try_from(self.bytes_written()).unwrap();
+            if s % PAGE_SZ == 0 {
+                s
+            } else {
+                s.checked_add(PAGE_SZ - (s % PAGE_SZ)).unwrap()
+            }
+        };
+        let buf = Vec::with_capacity(size);
+
+        // read from disk what we've already flushed
+        let writer = self.rw.as_writer();
+        let nwritten_blocks = usize::try_from(writer.nwritten_blocks).unwrap();
+        let buf = writer
```

Comment:
Kind of weird to have the main read function (`load_flushed_blocks_into_contiguous_memory`) as part of the writer type.  Maybe just give the writer a function that returns the information needed to execute the read?